poglink/cogs/rates.py

```diff
@@ -66,25 +66,33 @@ def __init__(self, client):
     async def get_current_rates(url):
         logger.debug(f"Requesting current server rates")
         async with aiohttp.ClientSession() as session:
-            async with session.get(
-                url,
-                headers={"Pragma": "no-cache", "Cache-Control": "no-cache"},
-            ) as response:
-                response = await response.text()
-                rates = RatesStatus.from_raw(response)
-                logger.debug(f"Obtained server rates: {RatesStatus}")
-                return rates
+            try:
+                async with session.get(
+                    url,
+                    headers={"Pragma": "no-cache", "Cache-Control": "no-cache"},
+                ) as response:
+                    response = await response.text()
+                    rates = RatesStatus.from_raw(response)
+                    logger.debug(f"Obtained server rates: {RatesStatus}")
+                    return rates
+            except ValueError as e:
+                raise RatesProcessError(e) from e
+            except Exception as e:
+                raise RatesFetchError(e) from e
 
     async def send_embed(self, description, url):
         # generate embed
         logger.debug(f"Attempting to send embed. desc: {description}, url: {url}")
         try:
-            server_match_dict = (
-                re.match(
-                    "(?P<host>.*\/)?(?:(?P<platform>.*)\_(?P<game_mode>.*)\_)?dynamicconfig\.ini",
-                    os.path.basename(urlparse(url).path),
-                )
-            ).groupdict()
+            match = re.match(
+                "(?P<host>.*\/)?(?:(?P<platform>.*)\_(?P<game_mode>.*)\_)?dynamicconfig\.ini",
+                os.path.basename(urlparse(url).path),
+            )
+            if match:
+                server_match_dict = match.groupdict()
+            else:
+                server_match_dict = {}
```

Comment:
I split up this regex because when writing tests with custom URLs I realized the match was returning as `None`, and therefore the `groupdict()` call was failling.